From 6167a55b227db61eb52c2a4f96f44fc559a8b1d0 Mon Sep 17 00:00:00 2001
From: Charles Haithcock <chaithco@redhat.com>
Date: Fri, 18 Jul 2025 16:14:25 -0600
Subject: [PATCH 2/5] doc: Update requirements for building on Fedora

Attempting to build on Fedora fails with the following error;

$ make
TARGET: RISCV64
 CRASH: 9.0.0++
   GDB: 16.2

Saving 'gdb-16.2.tar.gz'
[...]
checking for the correct version of gmp.h... no
configure: error: Building GDB requires GMP 4.2+, and MPFR 3.1.0+.
Try the --with-gmp and/or --with-mpfr options to specify
their locations.  If you obtained GMP and/or MPFR from a vendor
distribution package, make sure that you have installed both the libraries
and the header files.  They may be located in separate packages.
make[2]: *** No targets specified and no makefile found.  Stop.

crash build failed

make[1]: *** [Makefile:316: gdb_merge] Error 1
make: *** [Makefile:307: all] Error 2

Installing gmp-devel and mpfr-devel fixed this, so this patch updates the
requirements for building on Fedora.
---
 README | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README b/README
index 2e34fbb..f9824c7 100644
--- a/README
+++ b/README
@@ -73,7 +73,7 @@
      that is created in the top-level kernel build directory must be saved.
 
   o  Requirements for building:
-      Fedora: make gcc gcc-c++ ncurses-devel zlib-devel lzo-devel snappy-devel bison wget patch texinfo libzstd-devel
+      Fedora: make gcc gcc-c++ ncurses-devel zlib-devel lzo-devel snappy-devel bison wget patch texinfo libzstd-devel gmp-devel mpfr-devel
       Ubuntu/Debian: make gcc g++ libncurses-dev zlib1g-dev liblzo2-dev libsnappy-dev bison wget patch texinfo libzstd-dev
       Arch Linux: make gcc ncurses zlib lzo snappy bison wget patch texinfo zstd
       openSUSE: make gcc gcc-c++ ncurses-devel zlib-devel lzo-devel snappy-devel bison wget patch texinfo libzstd-devel
-- 
2.47.0

